src/ingest_validation_tools/validation_utils.py

```diff
@@ -249,65 +319,20 @@ def get_tsv_errors(
     # >>> errors = test_tsv('version\\tfake\\n1\\tfake', assay_type='codex')
     # >>> assert 'Unexpected fields' in errors[0]
     """
+    from ingest_validation_tools.upload import Upload
 
     logging.info(f"Validating {schema_name} TSV...")
-    if not Path(tsv_path).exists():
-        return {"File does not exist": f"{tsv_path}"}
-
-    try:
-        rows = dict_reader_wrapper(tsv_path, encoding=encoding)
-    except IsADirectoryError:
-        return {"Expected a TSV, but found a directory": f"{tsv_path}"}
-    except UnicodeDecodeError as e:
-        return {"Decode Error": get_context_of_decode_error(e)}
-
-    if not rows:
-        return {"File has no data rows": f"{tsv_path}"}
-
-    is_cedar = rows[0].get("metadata_schema_id")
-
-    version = "0"
-    # TODO: Fix this because we can't force the version to be 2 forever.
-    if is_cedar:
-        version = "2"
-    elif "version" in rows[0]:
-        version = rows[0]["version"]
-
-    if is_cedar:
-        from ingest_validation_tools.upload import Upload
 
-        upload = Upload(Path(tsv_path).parent, cedar_api_key=cedar_api_key)
-        errors = upload.api_validation(Path(tsv_path), report_type)
-        schema_version = SchemaVersion(schema_name, version)
-        return errors | upload._cedar_url_checks(str(tsv_path), schema_version)
-
-    try:
-        schema = get_schema_with_constraints(
-            schema_name, version, offline, optional_fields
-        )
-    except OSError as e:
-        return {e.strerror: Path(e.filename).name}
-
-    if schema.get("deprecated") and not ignore_deprecation:
-        return {"Schema version is deprecated": f"{schema_name}-v{version}"}
-
-    return get_table_errors(tsv_path, schema, report_type)
-
-
-def get_schema_with_constraints(
-    schema_name: str,
-    version: str,
-    offline: bool = False,
-    optional_fields: List[str] = [],
-) -> dict:
-    others = get_other_names()
-    if schema_name in others:
-        schema = get_other_schema(schema_name, version, offline=offline)
-    else:
-        schema = get_table_schema(
-            schema_name, version, offline=offline, optional_fields=optional_fields
-        )
-    return schema
+    # TODO: this is weird, because we're creating an upload for a single file...maybe subclass?
```

Comment:
Following up with the TODO, maybe this whole routine can go away since it was meant to be used as metadata.tsv stand-alone validator, and that happens on CEDAR?

Reply:
`get_tsv_errors` is called from Ingest API, so I preserved functionality here. It might be worth looking into whether we can change that though (perhaps outside of the scope of this PR?).